fili-core/src/main/java/com/yahoo/bard/webservice/web/SlicesApiRequest.java

```diff
@@ -165,7 +164,7 @@ protected SlicesApiRequest() {
             throw new BadApiRequestException(msg);
         }
 
-        Map<Column, List<Interval>> columnCache = table.getAvailability().getAvailableIntervals();
+        Map<Column, Set<Interval>> columnCache = table.getAllAvailableIntervals();
```

Comment:
By "less painful tests" I mean "tests that work on more than 1 system". The behavior we've seen is that passing things through a `HashSet` allows for ordering to be different on different platforms / machines, etc. and the tests that it breaks are often test where we're comparing an expected output JSON with an API response, and it's ordering of things within those JSON blobs that matters for comparison.

So, while I agree that we should generally not let the test needs force interface direction, consistent ordering is one of those places that we've not had much choice. If Java had an `OrderedSet` interface, we'd use that in a heart-beat, but it doesn't 😢